File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-swift/Sources/MailSdk/mailProviderCatalog.swift
public struct MailProviderCatalogEntry {
    public let providerKey: String
    public let pluginId: String
    public let driverId: String
    public let defaultSelected: Bool
}

public enum MailProviderCatalog {
    public static let DEFAULT_mail_PROVIDER_KEY: String = "smtp"

    public static let entries: [MailProviderCatalogEntry] = [
        .init(providerKey: "smtp", pluginId: "Mail-smtp", driverId: "sdkwork-mail-driver-smtp", defaultSelected: true),
        .init(providerKey: "imap", pluginId: "Mail-imap", driverId: "sdkwork-mail-driver-imap", defaultSelected: false),
    ]

public static func getMailProviderByProviderKey(_ providerKey: String) -> MailProviderCatalogEntry? {
        entries.first { $0.providerKey == providerKey }
    }

}

File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-kotlin/src/main/kotlin/com/sdkwork/Mail/metadata/mailProviderCatalog.kt
package com.sdkwork.Mail.metadata

data class MailProviderCatalogEntry(
    val providerKey: String,
    val pluginId: String,
    val driverId: String,
    val defaultSelected: Boolean,
)

object MailProviderCatalog {
    const val DEFAULT_mail_PROVIDER_KEY: String = "smtp"

    val entries: List<MailProviderCatalogEntry> = listOf(
        MailProviderCatalogEntry("smtp", "Mail-smtp", "sdkwork-mail-driver-smtp", true),
        MailProviderCatalogEntry("imap", "Mail-imap", "sdkwork-mail-driver-imap", false),
    )

fun getMailProviderByProviderKey(providerKey: String): MailProviderCatalogEntry? =
        entries.firstOrNull { it.providerKey == providerKey }

}
